### PR TITLE
Add ContentResolverTypeDefaults, allow default object mapping and Operations for type

### PR DESCRIPTION
**Pull request by artem-zinnatullin**
Part of #274 

@nikitin-da PTAL

### CODE REVIEW

File: storio-contentresolver/src/main/java/com/pushtorefresh/storio/contentresolver/operation/delete/PreparedDeleteObjects.java [hunk 1]
```diff
@@ -175,12 +146,24 @@ public Builder(@NonNull StorIOContentResolver storIOContentResolver, @NonNull It
...
             if (deleteResolver == null) {
-                deleteResolver = DefaultDeleteResolver.INSTANCE;
+                if (typeDefaults != null && typeDefaults.deleteResolver != null) {
+                    deleteResolver = typeDefaults.deleteResolver;
+                } else {
+                    deleteResolver = DefaultDeleteResolver.INSTANCE;
+                }
             }
 
+            checkNotNull(deleteResolver, "Please specify Delete Resolver");
```
Comment (nikitin-da) [inline storio-contentresolver/src/main/java/com/pushtorefresh/storio/contentresolver/operation/delete/PreparedDeleteObjects.java]: Is it possible?

Comment (artem-zinnatullin) [inline storio-contentresolver/src/main/java/com/pushtorefresh/storio/contentresolver/operation/delete/PreparedDeleteObjects.java]: Nope :) Just to be sure :D

Comment (nikitin-da) [inline storio-contentresolver/src/main/java/com/pushtorefresh/storio/contentresolver/operation/delete/PreparedDeleteObjects.java]: My psychiatrist said that I'm paranoid, I want to know has he said it someone else =))

Comment (nikitin-da) [inline storio-contentresolver/src/main/java/com/pushtorefresh/storio/contentresolver/operation/delete/PreparedDeleteObjects.java]: Well done.

**Pull request merged**